### DISCUSSION TITLE
How can I Spawn Objects using MIDI

### DISCUSSION THREAD

**Discussion by AceKraken35** [melanchall/drywetmidi#148] [Q&A] [answered]
Hi, I'm very sorry for this stupid question. I'm still a newb.
My question is how can I instantiate an object after reading the MIDI file.

**Comment by melanchall** [comment, marked as answer]
Can you please describe your task in more detail? Unfortunately I completely didn't understand what you need. What object you want to instantiate?

**Reply by AceKraken35** [reply]
For example I want to instantiate a moving cube continuously in sync with any notes being read in a midi file

**Reply by melanchall** [reply]
An average MIDI file will be read for tens of milliseconds. Obviously you don't want to do some animations during such a small amount of time. I suppose you want to **play** a MIDI file and do some actions (instantiate a moving cube or whatever) on every note while the file is playing, right?

**Reply by AceKraken35** [reply]
Well first is I don't want to play the MIDI, I just want to get the note data in it and also the time difference between each note. Then spawn an Object based on the data collected, is it possible? If it's not much can you give me an example so I can work my way out of it.

**Reply by melanchall** [reply]
You can get all notes from a MIDI file with this simple code:

```csharp
var notes = midiFile.GetNotes();
```

Each `Note` has `Time` and `Length` properties so you can easily calculate distances between them. Having all the file notes you can do whatever you want with them (spawn an object, setup animations and so on):

```csharp
foreach (var note in notes)
{
    // spawn an object
}
```
